examples/C++/DDS/BasicConfigurationExample/BasicConfigurationHelloWorldPublisher.h

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file BasicConfigurationHelloWorldPublisher.h
+ *
+ */
+
+#ifndef _EPROSIMA_FASTDDS_EXAMPLES_CPP_DDS_BASICCONFIGURATIONEXAMPLE_BASICCONFIGURATIONHELLOWORLDPUBLISHER_H_
+#define _EPROSIMA_FASTDDS_EXAMPLES_CPP_DDS_BASICCONFIGURATIONEXAMPLE_BASICCONFIGURATIONHELLOWORLDPUBLISHER_H_
+
+#include <atomic>
+#include <condition_variable>
+#include <mutex>
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/publisher/DataWriterListener.hpp>
+#include <fastdds/dds/topic/TypeSupport.hpp>
+
+#include "HelloWorldPubSubTypes.h"
+
+/**
+ * Class used to group into a single working unit a Publisher with a DataWriter, its listener, and a TypeSupport member
+ * corresponding to the HelloWorld datatype
+ */
+class HelloWorldPublisher
+{
+public:
+
+    HelloWorldPublisher();
+
+    virtual ~HelloWorldPublisher();
+
+    //!Initialize the publisher
+    bool init(
+            const std::string& topic_name,
+            uint32_t domain,
+            uint32_t num_wait_matched,
+            bool async,
+            const std::string& transport,
+            bool reliable,
+            bool transient);
+
+    //!Publish a sample
+    void publish();
+
+    //!Run for number samples, publish every sleep seconds
+    void run(
+            uint32_t number,
+            uint32_t sleep);
+
+    // returns the current state of execution
```

Comment:
**In general** starts the comments in upper case
**In general** use imperative voice  (`Return`)

Reply:
:+1: